trust_me_bro.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Creates a virtual environment if it does not exist
+if [ ! -d "venv" ] 
+then
+    python3 -m venv venv
+    echo "Virtual environment created."
+fi
+#pgrade pip
+pip install --upgrade pip
+
+# Activates the virtual environment
+source venv/bin/activate
+
+# Installs Poetry
+pip install poetry
+
+echo "Poetry has been installed."
+
+# Install dependencies using Poetry
+poetry install
+
+echo "Dependencies installed."
```

Comment:
**suggestion(non-blocking):** Run this script through a tool like shellcheck: https://www.shellcheck.net/

This will catch errors like:
Your script does not exit on error. If one of your commands eg. `python3 -m venv venv` returns non-zero status, your script will still try to execute next lines:
```
pip install --upgrade pip
source venv/bin/activate <- this will fail
pip install poetry <- this will install poetry somewhere (depends on your setup) but NOT in venv
```

Reply:
https://ioflood.com/blog/bash-exit-on-error/